### ISSUE TITLE
IntelliSense Client always shutting down

### ISSUE THREAD

**Issue by gabyx** [microsoft/vscode-cpptools#1908]
OS X, 0.16.1 C/C++ Extension , VS Code 1.22    

Intellisense Seems to work when VS Code is launched at the beginning, and always pretty soon
it hangs it self up ->
```bash
Shutting down IntelliSense server: ...ExecutionGraph/gui/executionGraphGui/cefapp/App.cpp
Failed to create IntelliSense client. No args provider
Checking for syntax errors: ...ExecutionGraph/gui/executionGraphGui/backend/BackendRequestHandler.cpp
Shutting down IntelliSense server: ...gui/executionGraphGui/cefapp/BackendResourceHandler.cpp
Failed to create IntelliSense client. No args provider
```
At this point nothing works anymore, formatting, code completiong...
Does anybody know where the problem is? What does `no args provider` mean?

**Comment by bobbrow** [comment]
That's a bug.  I can't think of how it's possible to get into this state, but we'll have to take another look at the code.  Does reloading the window fix it?  Does IntelliSense work for a while and then quit?

The args provider is just the object we use to aggregate arguments to send to the IntelliSense compiler.

**Comment by gabyx** [comment]
Reloading the window does not fix it, although Intellisense starts parsing again...
I have not found the way to reproduce it consistently. It seems that while looking up a definition or a symbol which is not parsed, Intellisense starts to dig, the blue progress bar moves along the top of the window and nothing happens anymore.
but that might be a very wrong guess...

**Comment by bobbrow** [comment]
I believe I found the root cause of this today.

**Comment by gabyx** [comment]
This is so good to hear!

**Comment by cento** [comment]
@bobbrow  share with us your suspicion

**Comment by bobbrow** [comment]
I could, but it probably wouldn't make sense since you don't have access to the code in question.  It looks like a boolean's intent got inverted when we made the fix to link header file changes to all translation units referencing them.

**Comment by cento** [comment]
Thanks for sharing it

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.17.5.
